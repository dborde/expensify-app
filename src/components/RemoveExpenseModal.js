import React from 'react';
import Modal from 'react-modal';

if (process.env.NODE_ENV !== 'test') Modal.setAppElement('#app');

const OptionModal = (props) => (
  <Modal
    isOpen={props.isModalOpen}
    onRequestClose={props.onToggleModal}
    contentLabel="Remove Expense Confirmation"
    closeTimeoutMS={200}
    className="modal"
  >
    <h3 className="modal__title">Remove Expense</h3>
    <p className="modal__body">Are you sure you want to remove <span>{props.expense.description}</span> expense?</p>
    <button className="button" onClick={props.onToggleModal}>Cancel</button>
    <button className="button button--caution" onClick={props.onRemove}>I&apos;m sure</button>
  </Modal>
);
export default OptionModal;
