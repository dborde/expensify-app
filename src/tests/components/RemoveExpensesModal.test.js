import React from 'react';
import { shallow } from 'enzyme';
import RemoveExpenseModal from '../../components/RemoveExpenseModal';
import expenses from '../fixtures/expenses';

let wrapper, isModalOpen, onRemove, onToggleModal, expense;

beforeEach(() => {
  isModalOpen = true;
  onRemove = jest.fn();
  onToggleModal = jest.fn();
  expense = expenses[0];
  wrapper = shallow(
    <RemoveExpenseModal
      expense={expense}
      isModalOpen={isModalOpen}
      onRemove={onRemove}
      onToggleModal={onToggleModal}
    />);
});

test('should render RemoveExpenseModal correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
