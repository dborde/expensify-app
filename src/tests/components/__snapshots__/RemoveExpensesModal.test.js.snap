// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RemoveExpenseModal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Remove Expense Confirmation"
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h3
    className="modal__title"
  >
    Remove Expense
  </h3>
  <p
    className="modal__body"
  >
    Are you sure you want to remove 
    <span>
      Gum
    </span>
     expense?
  </p>
  <button
    className="button"
    onClick={[Function]}
  >
    Cancel
  </button>
  <button
    className="button button--caution"
    onClick={[Function]}
  >
    I'm sure
  </button>
</Modal>
`;
